core/server/src/main/java/alluxio/worker/file/UnderFileSystemManager.java

```diff
@@ -285,7 +281,7 @@ private void cancel() throws IOException {
       mStream.close();
       UnderFileSystem ufs = UnderFileSystem.get(mUri);
       // TODO(calvin): Log a warning if the delete fails
-      ufs.delete(mTemporaryUri, false);
+      ufs.delete(mUri, false);
```

Comment:
Similar comment about adding a cancel API to the stream or otherwise working around this inefficiency.